blueflood-core/src/test/java/com/rackspacecloud/blueflood/io/serializers/CounterRollupSerializationTest.java

```diff
@@ -36,37 +29,25 @@
         BluefloodCounterRollup c1 = new BluefloodCounterRollup().withCount(34454722343L).withSampleCount(10);
-        
-        if (System.getProperty("GENERATE_COUNTER_SERIALIZATION") != null) {
-            OutputStream os = new FileOutputStream("src/test/resources/serializations/counter_version_" + Constants.VERSION_1_COUNTER_ROLLUP + ".bin", false);
-            os.write(Base64.encodeBase64(new NumericSerializer.CounterRollupSerializer().toByteBuffer(c0).array()));
-            os.write("\n".getBytes());
-            os.write(Base64.encodeBase64(new NumericSerializer.CounterRollupSerializer().toByteBuffer(c1).array()));
-            os.write("\n".getBytes());
-            os.close();
-        }
-        
-        Assert.assertTrue(new File("src/test/resources/serializations").exists());
-                
-        int count = 0;
-        int version = 0;
-        final int maxVersion = Constants.VERSION_1_COUNTER_ROLLUP;
-        while (version <= maxVersion) {
-            BufferedReader reader = new BufferedReader(new FileReader("src/test/resources/serializations/counter_version_" + version + ".bin"));
-            
-            ByteBuffer bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
-            BluefloodCounterRollup cc0 = NumericSerializer.serializerFor(BluefloodCounterRollup.class).fromByteBuffer(bb);
-            Assert.assertEquals(c0, cc0);
-            
-            bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
-            BluefloodCounterRollup cc1 = NumericSerializer.serializerFor(BluefloodCounterRollup.class).fromByteBuffer(bb);
-            Assert.assertEquals(c1, cc1);
-            
-            Assert.assertFalse(cc0.equals(cc1));
-            version++;
-            count++;
-        }
-        
-        Assert.assertTrue(count > 0);
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        baos.write(Base64.encodeBase64(new NumericSerializer.CounterRollupSerializer().toByteBuffer(c0).array()));
+        baos.write("\n".getBytes());
+        baos.write(Base64.encodeBase64(new NumericSerializer.CounterRollupSerializer().toByteBuffer(c1).array()));
+        baos.write("\n".getBytes());
+
+        ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
+        BufferedReader reader = new BufferedReader(new InputStreamReader(bais));
+
+        ByteBuffer bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
+        BluefloodCounterRollup cc0 = NumericSerializer.serializerFor(BluefloodCounterRollup.class).fromByteBuffer(bb);
+        Assert.assertEquals(c0, cc0);
+
+        bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
+        BluefloodCounterRollup cc1 = NumericSerializer.serializerFor(BluefloodCounterRollup.class).fromByteBuffer(bb);
+
+        Assert.assertEquals(c1, cc1);
+        Assert.assertFalse(cc0.equals(cc1));
```

Comment:
Yeah... I was getting lazy. I thought since they are just ByteArrayIn/OutputStream that closing them would not do much.